### PR TITLE
Code scan optimization

### PR DESCRIPTION
**Pull request by WenjiaoYue**
modify bandit.sh、pyspelling.sh、pylint.sh、code-scan.yml、lpot_dict.txt, add DockerfileCodeScan.devel.

add custom dockers for code-scan
print the log file contents to the output panel
enhance user experience with colored text prompts

### CODE REVIEW

File: .azure-pipelines/docker/DockerfileCodeScan.devel [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+RUN python -m pip install --no-cache-dir pylint==2.12.1\
+    bandit\
+    pyspelling\
+    google\
+    autograd\
+    ofa\
+    fvcore\
+    pymoo\
+    onnxruntime_extensions\
```
Comment (chensuyue) [inline .azure-pipelines/docker/DockerfileCodeScan.devel:37]: code scan required this component？

File: .azure-pipelines/docker/DockerfileCodeScan.devel [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    tf_slim\
+    transformers\
```
Comment (chensuyue) [inline .azure-pipelines/docker/DockerfileCodeScan.devel:39]: same question

File: .azure-pipelines/scripts/codeScan/bandit/bandit.sh [hunk 3]
```diff
@@ -1,17 +1,29 @@
 #!/bin/bash
-set -ex
+set -e
 
 mkdir -p /neural-compressor/.azure-pipelines/scripts/codeScan/scanLog
-
 bandit_log_dir="/neural-compressor/.azure-pipelines/scripts/codeScan/scanLog"
-pip install bandit
 
+# error
```
Comment (chensuyue) [inline .azure-pipelines/scripts/codeScan/bandit/bandit.sh:7]: Suggest to create a shell in the common scripts path, so everyone can export the shell when we want to use color.

Comment (chensuyue) [reply .azure-pipelines/scripts/codeScan/bandit/bandit.sh:7]: And the code will be more efficiency.

File: .azure-pipelines/docker/DockerfileCodeScan.devel (hunk 2 above)
Comment (WenjiaoYue) [reply .azure-pipelines/docker/DockerfileCodeScan.devel:39]: yeah! pylint requires these installation packages

File: .azure-pipelines/docker/DockerfileCodeScan.devel (hunk 1 above)
Comment (WenjiaoYue) [reply .azure-pipelines/docker/DockerfileCodeScan.devel:37]: it also pylint requires

File: .azure-pipelines/scripts/codeScan/bandit/bandit.sh (hunk 3 above)
Comment (WenjiaoYue) [reply .azure-pipelines/scripts/codeScan/bandit/bandit.sh:7]: okay!

File: .azure-pipelines/code-scan.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 trigger: none
 pr: 
-  - master
+  - code-scan
```
Comment (chensuyue) [inline .azure-pipelines/code-scan.yml:3]: forget to change back?

File: .azure-pipelines/code-scan.yml [hunk 5]
```diff
@@ -162,12 +159,10 @@ stages:
...
               script: |
+                source $(CURRENT_PATH)/.azure-pipelines/scripts/change_color.sh
                 set -e
-                mkdir -p $(CURRENT_PATH)/$(CODE_SCAN_LOG_PATH)
-                RED="echo -en \\E[1;31m"
-                LIGHT_PURPLE="echo -en \\E[35m"         
-                YELLOW="echo -en \\E[1;33m"       
-                RESET="echo -en \\E[0m \\n"
+                mkdir -p $(CURRENT_PATH)/$(CODE_SCAN_LOG_PATH)            
+                RESET="echo -en \\E[0m \\n"  # close
```
Comment (chensuyue) [inline .azure-pipelines/code-scan.yml:165]: what does this mean? I see the same variable in change_color file.

Comment (WenjiaoYue) [reply .azure-pipelines/code-scan.yml:165]: for the sake of beauty, code-scan added a newline at the exit,  change_color file is a common way of writing, so it is written in each script separately.

**Pull request merged**